pkg/controller/projectclaim/customresourceadapter.go

```diff
@@ -129,3 +129,29 @@ func (c *CustomResourceAdapter) EnsureProjectReferenceExists() error {
 	}
 	return nil
 }
+
+func (c *CustomResourceAdapter) EnsureProjectClaimState(state gcpv1alpha1.ClaimStatus) error {
+	if c.projectClaim.Status.State == state {
+		return nil
+	}
+
+	if state == gcpv1alpha1.ClaimStatusPending {
+		if c.projectClaim.Status.State != "" {
+			return nil
```

Comment:
Maybe we should return an error here as a unsupported state transition was requested.

Reply:
need to allow ClaimStatusPendingProject here, though